core/src/page_id.rs

```diff
@@ -128,13 +128,27 @@ impl PageId {
 
     /// Encode this page ID to its disambiguated (fixed-width) representation.
     pub fn encode(&self) -> [u8; 32] {
-        let mut uint = Uint::<256, 4>::from(0);
-        for limb in &self.path {
-            uint += Uint::from(limb + 1);
-            uint <<= 6;
-        }
+        if self.path.len() < 10 {
+            // 7 + 6*9 = 61 - max bit-width of a page with depth 9.
```

Comment:
It's not immediatelly obvious to me, where does this 7 come from?

Reply:
We do up to 9 shifts of a bit-pattern of 7 bits. The 7-bit pattern is `1000000` and is what you get when you add `1` to `111111`.